### PR TITLE
Init

### PR DESCRIPTION
**Pull request by ianaz**
Init for Proof of Existence node. 

This is a Proxeus external node, therefore the request (`/next`) should contain the parameter `tweetUrlOrID`. As the parameter says, this can be either a Tweet's url or an id only.

To run tests:
`make test`
`make build` will create artifacts, docker image
`make push` will push to proxeus' docker registry

To facilitate the review I put generation of fakes and go mod & sum in two separate commits so I suggest to check only last one

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+init:
```
Comment (alexblockfactory) [inline Makefile:8]: is there a reason not to use the same makefile as in other node implementations, like e.g.: https://github.com/ProxeusApp/node-mail-sender/blob/master/Makefile

File: README.md [hunk 2]
```diff
@@ -1,2 +1,50 @@
...
-An external node implementation for Proxeus core. Returns tweet's information given an id/url 
+An external node implementation for Proxeus core. Returns social posts information given an id or url 
+
+## Implementation
+
+Current implementation only supports Twitter posts. 
+
+## Usage
+
+We suggest to start it using docker.
```
Comment (alexblockfactory) [inline README.md]: "It is recommended"?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,50 @@
...
+
+Our latest image is available at `proxeus/node-proof-of-existence:latest`
```
Comment (alexblockfactory) [inline README.md]: "The latest image"?

**Review by alexblockfactory — APPROVED** [review]
lgtm. however the building the image locally with "make build" and then running it does not work on osx. we'd need `GOOS=linux GOARCH=arm` as in other node repos

File: Makefile (hunk 1 above)
Comment (ianaz) [reply Makefile:8]: changed

File: README.md (hunk 2 above)
Comment (ianaz) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (ianaz) [reply README.md]: done

**Comment by ianaz** [comment]
Changed the makefile, should work now

File: .gitignore [hunk 4]
```diff
@@ -11,5 +11,4 @@
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
 
-# Dependency directories (remove the comment below to include it)
-# vendor/
+./main
```
Comment (alexblockfactory) [inline .gitignore]: now its in artifacts/node-proof-of-existence

**Comment by alexblockfactory** [comment]
works on osx

File: .gitignore (hunk 4 above)
Comment (ianaz) [reply .gitignore]: What an eye! :) changed in d5b5a8b

**Pull request merged**